File: Python/bubble-sort.py

def bubbleSort(arr): 
    n = len(arr) 
  
    # Traverse through all  elements 
    for i in range(n): 
  
         
        for j in range(0, n-i-1): 
  
            
            # Swap if the element found is greater 
            # than the next element 
            if arr[j] > arr[j+1] : 
                arr[j], arr[j+1] = arr[j+1], arr[j] 

File: Ruby/bubble-sort.rb
class Main
  def self.run
    arr = [93, 2, 47, 1, 902]
    puts bubble_sort(arr)
  end

  def self.bubble_sort(arr)
    (0).upto((arr.length - 2)).each do |i|
      (0).upto((arr.length - 2 - i)).each do |j|
          arr[j], arr[j+1] = arr[j+1], arr[j] if arr[j] > arr[j+1]
      end
    end
    arr
  end
end


Main.run